cmd/ci-secret-generator/README.md

```diff
@@ -1,13 +1,40 @@
 # CI-Secret-Generator
 
-This tool aims to automate the process of deployment of secrets to Bitwarden while also providing a platform to document the commands used to generate these secrets. 
+This tool automates the process of deployment of secrets while also providing a
+platform to document the commands used to generate them.
 
-## Args and config.yaml
+## Target
 
-The tool expects a configuration like the one below which specifies the mapping between the `itemName`+`attributeName`/`attachmentName`/`fieldName` and the command used to generate the secret.
-The output of the command is stored into Bitwarden as the contents of the field/attachment/password.
+Different output targets are supported, which can be specified with the
+`--target` command-line option.
 
-`password` is the only valid name which is accepted for an attribute
+### `validate`
```

Comment:
Not sure if `validate` really makes sense here, as it isn't mutually exclusive with the other targets

Reply:
And the other targets are probably more of an output or a destination. Also, bitwarden/vault can be inferred from what credentials get passed. File probably as well?

Reply:
> Not sure if validate really makes sense here, as it isn't mutually exclusive with the other targets

One of the many reasons for the change is that there are *two* arguments related to validation.  `--validate-only` is what is being substituted by `--target validate`, and is a mode of execution that stops after the configuration is loaded.  `--validate` is a separate argument (which is kept in this PR) that can be used with any other mode and validates the generator's configuration against `ci-secret-bootstrap`'s configuration.

Amusingly, `--validate-only --validate=false` is a possible combination which does only the implicit (generator) configuration validation when it is loaded.

Reply:
> And the other targets are probably more of an output or a destination. Also, bitwarden/vault can be inferred from what credentials get passed. File probably as well?

That is roughly how the tool already works, but I tried to make the interface a bit more explicit.  A comparison of the before/after combination of arguments (partially described in the commit message):

Validate bootstrap/generator configuration files:

`ci-secret-generator --config "$config" --bootstrap-config "$bootstrap_config" --validate-only`

`ci-secret-generator --config "$config" --bootstrap-config "$bootstrap_config" --target validate`

Dry-run mode (the default), output to a file:

`ci-secret-generator --config "$config" --output-file /dev/stdout`

`ci-secret-generator --config "$config" --target file --output-file /dev/stdout`

Bitwarden:

`ci-secret-generator --config "$config" --dry-run=false --bw-user test --bw-password-path /dev/null`

`ci-secret-generator --config "$config" --target bitwarden --bw-user test --bw-password-path /dev/null`

Vault:

`N/A`

`ci-secret-generator --config "$config" --target vault --vault-addr http://localhost:8200 --vault-token /dev/null --vault-prefix prefix`

Reply:
yeah but all of file, bitwarden, and vault can be derived from other options, so the "more explicit" here means "redundant". As a user, I expect that after I pass credentials for Vault it will write to Vault and not also expect me to set another option. I agree that the `--validate-only` is a bit misleading, technically its more of a `--dont-write-to-secretstore`